stripe/src/main/java/com/stripe/android/IssuingCardPinService.java

```diff
@@ -0,0 +1,268 @@
+package com.stripe.android;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import com.stripe.android.exception.APIConnectionException;
+import com.stripe.android.exception.APIException;
+import com.stripe.android.exception.AuthenticationException;
+import com.stripe.android.exception.CardException;
+import com.stripe.android.exception.InvalidRequestException;
+
+import org.json.JSONException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/*
+ * Methods for retrieval / update of a Stripe Issuing card
+ */
+public class IssuingCardPinService
+        implements EphemeralKeyManager.KeyManagerListener<IssuingCardEphemeralKey> {
+
+    private static final long KEY_REFRESH_BUFFER_IN_SECONDS = 30L;
+    private static final String PIN_RETRIEVE = "PIN_RETRIEVE";
+    private static final String PIN_UPDATE = "PIN_UPDATE";
+
+    private static IssuingCardPinService mInstance;
+    private @NonNull
```

Comment:
nit: put annotation before `private` (we should create a checkstyle for this)

Reply:
oh yeah